kv/txn_test.go

```diff
@@ -151,6 +151,7 @@ func verifyUncertainty(concurrency int, maxOffset time.Duration, t *testing.T) {
 
 	// Initial high offset to allow for future writes.
 	s.Clock.SetMaxOffset(999 * time.Nanosecond)
+	s.Manual.Set(s.Clock.MaxOffset().Nanoseconds() + 1)
```

Comment:
Tobias, this is necessary because now when we grant the leader lease, the low water mark on the timestamp cache is set to 0 + maxOffset. Setting the clock preemptively here to a value higher than the max offset keeps us from running into the low water mark on writes unnecessarily.